File: Shopiall/Models/UpsellModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Shopiall.Models
{
    public class UpsellModel
    {
        public string Title { get; set; }

        public string PrimaryText { get; set; }

        public string SecondaryText { get; set; }

        public long[] ProductIds { get; set; }
    }
}

File: Shopiall/ClientApp/src/app/features/upsell/upsell.model.ts
import { ColumnDef } from '../../shared/models/table.model';
import { Product } from '../product/product.model';

export interface Upsell {
  id: string;
  title: string;
  primaryText: string;
  secondaryText: string;
  productIds: string;
  products: Product[];
}

export const colDefs = [
  {
    field: 'title',
    headerName: 'Titre',
    checkboxSelection: true,
    resizable: true,
  } as ColumnDef,
  {
    field: 'primaryText',
    headerName: 'Texte principal',
  } as ColumnDef,
  {
    field: 'secondaryText',
    headerName: 'Texte secondaire',
  } as ColumnDef,
  {
    field: 'ownerProduct',
    headerName: 'Belong to',
  } as ColumnDef,
  {
    field: 'productIds',
    headerName: 'With products',
  } as ColumnDef,
];
